core-web/libs/sdk/experiments/src/lib/components/withExperiments.tsx

```diff
@@ -29,11 +31,22 @@ export const withExperiments = (
     WrappedComponent: React.ComponentType<DotcmsPageProps>,
     config: DotExperimentConfig
 ) => {
-    return (props: DotcmsPageProps) => {
-        return (
-            <DotExperimentsProvider config={{ ...config }}>
-                <DotExperimentHandlingComponent {...props} WrappedComponent={WrappedComponent} />
-            </DotExperimentsProvider>
-        );
-    };
+    // We need to use a custom memoization hook
+    // because the useMemo or React.memo lose the reference of the object
+    // in each render, causing the experiment handling to be reinitialized.
+    const memoizedConfig = useMemoizedObject(config);
+
+    return useCallback(
+        (props: DotcmsPageProps) => {
+            return (
+                <DotExperimentsProvider config={memoizedConfig}>
+                    <DotExperimentHandlingComponent
+                        {...props}
+                        WrappedComponent={WrappedComponent}
+                    />
+                </DotExperimentsProvider>
+            );
+        },
+        [WrappedComponent, memoizedConfig]
+    );
```

Comment:
What if we use a useEffect and a useState? I cannot remember if we tested it

```javascript
export const useDotcmsLayoutWithExperiments = (
    config: DotExperimentConfig
) => {
    const [DotcmsLayoutComponent, setDotcmsLayoutComponent] =
        useState<React.ComponentType<DotcmsPageProps>>(DotcmsLayout);

    useEffect(() => {

        if(LOGIC FOR THE CONFIG TO PASS) {
            setDotcmsLayoutComponent((props: DotcmsPageProps) => {
                return (
                    <DotExperimentsProvider config={{ ...config }}>
                        <DotExperimentHandlingComponent {...props} WrappedComponent={DotcmsLayout} />
                    </DotExperimentsProvider>
                );
            };);

        }


    }, [config]);


    return DotcmsLayoutComponent;
};

```

Reply:
This way we dont have to do any logic in the example, just call the hook and use the component it returns, that can be `DotcmsLayout` or the Experiments one

Reply:
We tried and it didn't work :(
Although we didn't test exactly what you suggest, since sdk-experiments doesn't have sdk-react as a dependency
I'm not sure if we want to apply that logic on our end, I think a separate SDK was created for the client to handle.
But I'm not closed to the idea.
Thoughts? @fmontes @oidacra

Reply:
We tested something like this, but we didn't encapsulate the value inside a `useState`, you can try it recieving the `WrappedComponent` I just did it like that to avoid that extra logic

Reply:
I just tried it like this (Passing `WrappedComponent`), and we have the same problem 
I think it is because each render creates a new component
Even if we are saving references and putting deps in useEffect, in each render it is a new component, so it is lost

Reply:
Alright. This code just seems odd for me. But we can leave it like this in the mean time